tests/test_schema.py

```diff
@@ -0,0 +1,253 @@
+import uuid
+
+import freezegun
+import pytest
+
+from django.core.exceptions import SuspiciousOperation, ValidationError
+from django.db import (
+    DEFAULT_DB_ALIAS,
+    InternalError,
+    ProgrammingError,
+    connection,
+)
+from psycopg2 import errorcodes
+
+from psqlextra.error import extract_postgres_error
+from psqlextra.schema import PostgresSchema, postgres_temporary_schema
+
+
+def _does_schema_exist(name: str) -> bool:
+    with connection.cursor() as cursor:
+        return name in connection.introspection.get_schema_list(cursor)
+
+
+def test_postgres_schema_create():
+    schema = PostgresSchema.create("myschema")
+    assert schema.name == "myschema"
+    assert schema.using == DEFAULT_DB_ALIAS
+
+    assert _does_schema_exist(schema.name)
+
+
+def test_postgres_schema_does_not_overwrite():
+    schema = PostgresSchema.create("myschema")
+
+    with pytest.raises(ProgrammingError):
+        PostgresSchema.create(schema.name)
+
+
+def test_postgres_schema_create_max_name_length():
+    with pytest.raises(ValidationError):
+        PostgresSchema.create(
+            "stringthatislongerhtan63charactersforsureabsolutelysurethisislongerthanthat"
+        )
+
+
+def test_postgres_schema_create_name_that_requires_escaping():
+    # 'table' needs escaping because it conflicts with
+    # the SQL keyword TABLE
+    schema = PostgresSchema.create("table")
+    assert schema.name == "table"
+
+    assert _does_schema_exist("table")
+
+
+def test_postgres_schema_create_random():
+    with freezegun.freeze_time("2023-04-07 13:37:00.0"):
+        schema = PostgresSchema.create_random("myprefix")
+
+    assert schema.name == "myprefix_2023040713041680892620"
+    assert _does_schema_exist(schema.name)
+
+
+def test_postgres_schema_delete_and_create():
+    schema = PostgresSchema.create("test")
+
+    with connection.cursor() as cursor:
+        cursor.execute("CREATE TABLE test.bla AS SELECT 'hello'")
+        cursor.execute("SELECT * FROM test.bla")
+
+        assert cursor.fetchone() == ("hello",)
+
+    # Should refuse to delete since we added a table to the schema
+    with pytest.raises(InternalError) as exc_info:
+        schema = PostgresSchema.delete_and_create(schema.name)
+
+    pg_error = extract_postgres_error(exc_info.value)
+    assert pg_error.pgcode == errorcodes.DEPENDENT_OBJECTS_STILL_EXIST
```

Comment:
A test for the table would be nice here. Unlikely to get an error while the schema is still being deleted but the test is cheap.

Reply:
Added